src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Serialization/ObjectWriter.cs

```diff
@@ -251,11 +251,44 @@ public void WriteValue(object? value)
         }
 
         /// <summary>
-        /// Write an array of bytes. The array data is provided as a
-        /// <see cref="ReadOnlySpan{T}">ReadOnlySpan</see>&lt;<see cref="byte"/>&gt;, and deserialized to a byte array.
+        /// Write an array of bytes. The array data is provided as a <see
+        /// cref="ReadOnlySpan{T}">ReadOnlySpan</see>&lt;<see cref="byte"/>&gt;, and deserialized to a byte array.
         /// </summary>
         /// <param name="span">The array data.</param>
-        public void WriteValue(ReadOnlySpan<byte> span)
+        public void WriteSpan(ReadOnlySpan<byte> span)
```

Comment:
>public void WriteSpan(ReadOnlySpan<byte> span) [](http://example.com/codeflow?start=8&length=46)

Just going to mention it in the PR for posterity, but would be nice if we wouldn't need to write the preamble for every chunk we want persisted.